### DISCUSSION TITLE
Auto-tagging from nested directory tree structure

### DISCUSSION THREAD

**Discussion by kulak** [stumpapp/stump#522] [Ideas]
It would be nice if tags were assgined from directory tree structure or from file in a directory tree structure.

For example, if I have path to book:  

```
Fiction -> SciFi -> Author -> Series Name  -> File with Book
```

Then if stump picks tags from directory names, we will have tags:

- Fiction
- SciFi
- Author
- Series Name

seems nice. 

Because stump offers multiple series tree types, the above may not work.  In this case, sutmp metadata file `stump.yml` in any directory, would allow us to provide desired tags for books in the directory:

`stump.yml`:

```yml
tags:
- Fiction
```

That way one could provide tags only for category directories Fiction and SciFi and end result will have tags:

- Fiction
- SciFi

It would allow to have specific categories without having overly generic.  One could provide `stump.yml` only in SciFi directory.

I used Yaml file as example. It could be TOML as it is supposed to be easier on the user.

**Comment by aaronleopold** [comment]
> Because stump offers multiple series tree types, the above may not work

Yeah I don't see a way to make this work without serious time and effort to reconcile both file-based systems

I'd be more amenable towards a DB-first approach, following the same direction of ignore rules (originally defined in a `.stumpignore` file, now it is configured in the UI).

That being said, Stump already supports [`series.json`](https://github.com/mylar3/mylar3/wiki/series.json-schema-%28version-1.0.1%29) files. I'd be open to extending away from the standard spec to intake tags there, as well. Since JSON is fairly widely used for this, though, I'd be hesitant to change format to TOML (even though I also like the format)
